flow/snapshot.go

```diff
@@ -2,25 +2,54 @@ package flow
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 )
 
 type SnapshotService interface {
+	Get(ctx context.Context, snapshotId Id) (*Snapshot, *Response, error)
 	List(ctx context.Context, options PaginationOptions) ([]*Snapshot, *Response, error)
 	Create(ctx context.Context, data *SnapshotCreate) (*Snapshot, *Response, error)
 	Delete(ctx context.Context, snapshotId Id) (*Response, error)
 }
 
+type SnaphostStatusKey = string
+
+const (
+	SnapshotStatusKeyAvailable SnaphostStatusKey = "available"
+	SnapshotStatusKeyCreating  SnaphostStatusKey = "creating"
+	SnapshotStatusKeyError     SnaphostStatusKey = "error"
+)
+
+type Status struct {
+	Id   Id                `json:"id"`
+	Name string            `json:"name"`
+	Key  string `json:"key"`
+}
+
 type Snapshot struct {
 	Id        Id       `json:"id"`
 	Name      string   `json:"name"`
 	Size      int      `json:"size"`
+	Status    Status   `json:"status"`
 	Volume    Volume   `json:"volume"`
 	Product   Product  `json:"product"`
 	CreatedAt DateTime `json:"created_at"`
 }
 
+func (snapshot *Snapshot) IsAvailable() (bool, error) {
+	switch snapshot.Status.Key {
+	case SnapshotStatusKeyAvailable:
+		return true, nil
+	case SnapshotStatusKeyCreating:
+		return false, nil
+	case SnapshotStatusKeyError:
+		return false, errors.New(fmt.Sprintf("Snapshot with id %d has errored state", snapshot.Id))
+	}
+	return false, errors.New(fmt.Sprintf("Snapshot with id %d has unknown status '%s'", snapshot.Id, snapshot.Status.Key))
+}
```

Comment:
why not simply use the following?

```suggestion
func (s *Snapshot) IsAvailable() bool {
    return s.Status == SnapshotStatusKeyAvailable
}
```

Reply:
Was thinking that we could extend this with other statuses which might evaluate to an available state thats why I added the switch case. For example if we would have a status for post-processing which won't prevent the snapshot from being used to create a new volume in the future but I guess we can add that when we need it.

Reply:
alright, yeah but then i'd suggest using a map like
```go
var availabilityMapping = map[SnaphostStatusKey]bool{
  SnapshotStatusKeyAvailable: true,
  SnapshotStatusKeyCreating: false,
}

func (s *Snapshot) IsAvailable() bool {
  return availabilityMapping[s.Status.Key]
}
```